test/performance/tools/common.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Setup env vars
+export PROJECT_NAME="knative-gcp-performance"
+export USER_NAME="mako-job@knative-gcp-performance.iam.gserviceaccount.com"
+export TEST_ROOT_PATH="$GOPATH/src/github.com/google/knative-gcp/test/performance"
+export KO_DOCKER_REPO="gcr.io/knative-gcp-performance"
+export TEST_CONFIG_VARIANT="continuous"
+export TEST_NAMESPACE="default"
+export PUBSUB_SERVICE_ACCOUNT="performance-pubsub-test"
+export PUBSUB_SERVICE_ACCOUNT_KEY="$(mktemp)"
+export PUBSUB_SECRET_NAME="google-cloud-key"
+
+function header() {
+  echo "***** $1 *****"
+}
+
+# Exit script, dumping current state info.
+# Parameters: $1 - error message (optional).
+function abort() {
+  [[ -n $1 ]] && echo "SCRIPT ERROR: $1"
+  exit 1
+}
+
+# Creates a new cluster.
+# $1 -> name, $2 -> zone/region, $3 -> num_nodes
+function create_cluster() {
+  header "Creating cluster $1 with $3 nodes in $2"
+  gcloud beta container clusters create ${1} \
+    --addons=HorizontalPodAutoscaling,HttpLoadBalancing \
+    --machine-type=n1-standard-4 \
+    --cluster-version=latest --region=${2} \
+    --enable-stackdriver-kubernetes --enable-ip-alias \
+    --num-nodes=${3} \
+    --enable-autorepair \
+    --scopes cloud-platform
+}
+
+# Create serice account secret on the cluster.
+# $1 -> cluster_name, $2 -> cluster_zone
+function create_secret() {
+  echo "Create service account on cluster $1 in zone $2"
+  gcloud container clusters get-credentials $1 --zone=$2 --project=${PROJECT_NAME} || abort "Failed to get cluster creds"
+  kubectl create secret generic service-account --from-file=robot.json=${PERF_TEST_GOOGLE_APPLICATION_CREDENTIALS}
+}
+
+# Set up the user credentials for cluster operations.
```

Comment:
I'm not sure whether I need to set up pubsub related service accounts and secrets in the function "setup_user" or not